File: core/src/org/rexcellentgames/burningknight/debug/ZoomCommand.java
package org.rexcellentgames.burningknight.debug;

import com.badlogic.gdx.Gdx;
import org.rexcellentgames.burningknight.Display;

public class ZoomCommand extends ConsoleCommand {
	{
		name = "/zoom";
		shortName = "/z";
	}

	@Override
	public void run(Console console, String[] args) {
		if (args.length == 0) {
			return;
		}

		float zoom = Math.max(0, Float.valueOf(args[0]));
		Gdx.graphics.setWindowedMode((int) (Display.GAME_WIDTH * zoom), (int) (Display.GAME_HEIGHT * zoom));
	}
}

File: core/src/org/rexcellentgames/burningknight/debug/ZoomCommand.kt
package org.rexcellentgames.burningknight.debug


import com.badlogic.gdx.Gdx
import org.rexcellentgames.burningknight.Display

class ZoomCommand : ConsoleCommand("/zoom", "/z", "Zoom [int]") {

	override fun run(console: Console, args: Array<String>) {
		if (args.isEmpty()) {
			return
		}

		val zoom = Math.max(0, Integer.valueOf(args[0]))

		Gdx.graphics.setWindowedMode(Display.GAME_WIDTH * zoom, Display.GAME_HEIGHT * zoom)
	}
}